From 3187389d787dce7185c24a31ca0c166de3ac9674 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Tue, 24 Sep 2019 13:56:02 +0800
Subject: [PATCH] PONRTSYS-5320/PONRTSYS-5319: CQM Timeout Configuration for
 EPON & GPON

PIB_CTRL.DQ_DLY register default value of 6 is too small to bridge short interruptions in the dequeue proccess which can be caused by DDR refresh cycles. So The timeout value shall therefore be changed to 7 for GPON and 0xF for EPON
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index 45bc8cf7c6a0..a4474579746c 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -1974,9 +1974,11 @@ static s32 do_port_setting(u32 *pmac, u32 flags, u32 *cbm_port,
 		ctrl.pib_bypass = 0;
 		ctrl.pib_en = 1;
 		ctrl.wakeup_intr_en = 1;
+		ctrl.deq_delay = 7;
 		config_pib_ctrl(&ctrl,
 				FLAG_PIB_BYPASS |
 				FLAG_PIB_ENABLE |
+				FLAG_PIB_DELAY |
 				FLAG_PIB_WAKEUP_INTR);
 		if (cqm_ctrl->force_xpcs)
 			set_ifmux(PRX300_WAN_AON_MODE);
@@ -1992,9 +1994,11 @@ static s32 do_port_setting(u32 *pmac, u32 flags, u32 *cbm_port,
 		ctrl.pib_en = 1;
 		ctrl.wakeup_intr_en = 1;
 		ctrl.pkt_len_adj = 1;
+		ctrl.deq_delay = 0xf;
 		config_pib_ctrl(&ctrl,
 				FLAG_PIB_BYPASS |
 				FLAG_PIB_ENABLE |
+				FLAG_PIB_DELAY |
 				FLAG_PIB_WAKEUP_INTR |
 				FLAG_PIB_PKT_LEN_ADJ);
 		if (cqm_ctrl->force_xpcs)
